src/mapping/impl/BasisFunctions.hpp

```diff
@@ -3,30 +3,40 @@
 #if defined(__NVCC__)
 
 #include <cuda_runtime.h>
+#include <ginkgo/extensions/kokkos.hpp>
+#include <ginkgo/ginkgo.hpp>
+
 #define BOOST_PP_VARIADICS 1
 #define PRECICE_HOST_DEVICE __host__ __device__
 #define PRECICE_MEMORY_SPACE __device__
-#define FMA fma
+#define FMA Kokkos::fma
+#define LOG Kokkos::log
 
 #elif defined(__HIPCC__)
 
-// #define __HIP_PLATFORM_AMD__
+#include <ginkgo/extensions/kokkos.hpp>
+#include <ginkgo/ginkgo.hpp>
 #include <hip/hip_runtime.h>
+
 #define PRECICE_HOST_DEVICE __host__ __device__
 #define PRECICE_MEMORY_SPACE __device__
-#define FMA fma
+#define FMA Kokkos::fma
+#define LOG Kokkos::log
 
 #else
 
 #define PRECICE_HOST_DEVICE
 #define PRECICE_MEMORY_SPACE
 #define FMA std::fma
+#define LOG std::log
```

Comment:
Is there any reason for this aside convenience?

Macros need to be prefixed due to xSDK, which eliminates the length upside.

Reply:
Yes, the macro maps to different functions, depending on which compiler looks at the file. Either the host compiler, or the kokkos launcher which might select the device compiler. For device code, we make use of `Kokkos::log` and `Kokkos::fma`.

Reply:
Given that function aliases are a pain, and selective overloading is too cumbersome, I would either create a wrapper `precice::mapping::detail::log(double d)` which internally calls the one or other, or prefix the macro.

Reply:
> or prefix the macro.

..and undefine it